proxy/envoy/config.go

```diff
@@ -184,13 +184,28 @@ func buildOutboundFilters(instances []*model.ServiceInstance, services []*model.
 	suffix := sharedInstanceHost(instances)
 	httpConfigs := make(HTTPRouteConfigs)
 
+	// get all the route rules applicable to the instances
+	rules := config.RouteRulesBySource("", instances)
```

Comment:
It's in a pod. If there are two services pointing to the pod, or two services with distinct ports, we'd get two elements in the array, wouldn't we?